File: Java/35_SearchInsertPosition/Solution.java
import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.stream.Collectors;

public class Solution {
  public int searchInsert(int[] numbers, int target) {
    if (numbers.length == 0) return 0;
    List<Integer> list = Arrays.stream(numbers).boxed().collect(Collectors.<Integer>toList());
    for (int i = 0; i < list.size(); i++)
      if(list.get(i) == target) return i;
    list.add(target);
    Collections.sort(list);
    for (int i = 0; i < list.size(); i++)
      if(list.get(i) == target) return i;
    return -1;
  }
  public static void main(String[] args) {
    int[] nums = {1,3,5,6};
    int target = 7;
    Solution solution = new Solution();
    System.out.println(solution.searchInsert(nums,target));
      
  }
}

File: C++/35_SearchInsertPosition/Solution.cpp
#include<iostream>
#include<vector>
#include<algorithm>

int searchInsert(std::vector<int>& nums, int target) {
  if (nums.size() == 0) return 0;
  for( int i = 0; i < nums.size(); i++)
    if (nums[i] == target ) return i;
  nums.push_back(target);
  std::sort(nums.begin(), nums.end());
  for( int i = 0; i < nums.size(); i++)
    if (nums[i] == target ) return i;
  return -1;
}

int main ( void ) {
  std::vector<int> numbers = {1,3,5,6};
  int target = 5;
  std::cout<<searchInsert(numbers, target)<<std::endl;

  return 0;
}